arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,268 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateLoadReportViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The table view to display the load reports.
+    @IBOutlet var tableView: UITableView!
+    
+    // MARK: Properties
+    
+    /// A feature service for an electric utility network in Naperville, Illinois.
+    let utilityNetwork = AGSUtilityNetwork(url: URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!)
+    /// The utility element to start the trace from.
+    var startingLocation: AGSUtilityElement?
+    /// The initial conditional expression.
+    var initialExpression: AGSUtilityTraceConditionalExpression?
+    /// The trace parameters for creating load reports.
+    var traceParameters: AGSUtilityTraceParameters?
+    /// The network attributes for the comparison.
+    var phasesNetworkAttribute: AGSUtilityNetworkAttribute?
+    /// A list of possible phases populated from the network's attributes.
+    var phaseChoices = [AGSCodedValue]()
+    /// A list of phases for which to create load reports.
+    var phaseSummaries = [(phase: AGSCodedValue, summary: PhaseSummary?)]()
+    
+    /// A struct for the phase summary, which contains the phase,
+    /// the total customers and total load for the phase.
+    struct PhaseSummary {
+        let phase: AGSCodedValue
+        let totalCustomers: Int
+        let totalLoad: Int
+        
+        var description: String {
+            "Customers: \(totalCustomers)\tLoad: \(totalLoad)"
```

Comment:
For user-visible text, it is best not to use string interpolation (unless you're using SwiftUI, but that's a totally different thing). String interpolation doesn't produce a localized result, so it is not a practice to model for our customers who may localize their apps. Instead, I would suggest using a number formatter. Then the number of customers can be displayed with a grouping separator ("1,190" instead of "1190"), which is what I expected. What does `Load` represent? I don't know if it makes sense for that value to use a grouping separator.

Since `description` is only used by the table view, it might make sense to eliminate this property and just construct the string in `tableView(_:cellForRowAt:)`.

Reply:
I have a question here: what is the best practice to construct strings with both mutable and immutable parts?

In our case for example, I want to display the string "Phase: \(phase.name)" in the cell's title text label. I know it is the best to use a static label to display the "Phase: " portion, but since I don't want to do too much UI work here, what about sth like `let title = "Phase: " + phase.name`? So that the first string can still be localized.